crates/shared/src/sources/uniswap_v3/pool_fetching.rs

```diff
@@ -87,25 +93,34 @@ impl TryFrom<PoolData> for PoolInfo {
     }
 }
 
-pub struct CachedPool {
-    pub pool: PoolData,
-    pub updated_at: Instant,
-    pub requested_at: Instant,
+#[derive(Default)]
+struct PoolsCheckpoint {
+    /// Pools state checkpoint on the `last_checkpoint_block` number.
+    data: HashMap<H160, PoolData>,
+    /// Last block considered reorg safe.
+    last_checkpoint_block: Option<u64>,
```

Comment:
Seems a bit inconsistent that a `PoolsCheckpoint` might not have its `last_checkpoint_block` set. Could it be a bit nicer to work around the problem of instantiating now and initializing later by making `pools_checkpoint: Mutex<Option<PoolsCheckpoint>>` instead?

Reply:
Resolved with initialization being mandatory, therefore last_checkpoint_block exist always.